django/db/models/base.py

```diff
@@ -1120,6 +1120,14 @@ def _save_table(
             and (
                 (meta.pk.default and meta.pk.default is not NOT_PROVIDED)
                 or (meta.pk.db_default and meta.pk.db_default is not NOT_PROVIDED)
+                or (
+                    meta.is_composite_pk
+                    and all(
+                        (f.default and f.default is not NOT_PROVIDED)
+                        or (f.db_default and f.db_default is not NOT_PROVIDED)
+                        for f in meta.pk
+                    )
+                )
             )
```

Comment:
As brought up in the other comment it seems like both the falsey defaults and composite primary key support can be addressed by doing

```suggestion
            and all(
                (f.default is not None and f.default is not NOT_PROVIDED)
                or (f.db_default is not None and f.db_default is not NOT_PROVIDED)
                for f in meta.pk_fields
            )
```